### ISSUE TITLE
How to use `YAMLRegExpTree` with XML configuration

### ISSUE THREAD

**Issue by hkrutzer** [ClickHouse/ClickHouse#60951]
https://clickhouse.com/docs/en/sql-reference/dictionaries#use-regular-expression-tree-dictionary-in-clickhouse-open-source describes how to create a `YAMLRegExpTree` from SQL, but what if I want to use it from XML definition as described [here](https://clickhouse.com/docs/en/sql-reference/dictionaries#configuring-a-dictionary)? The file source requires something like:

```xml
<source>
  <file>
    <path>/opt/dictionaries/os.tsv</path>
    <format>TabSeparated</format>
  </file>
</source>
```

but the YAMLRegExpTree has YAML format which is not an allowed format. Leaving out the format does not work.
In SQL, no format is defined:
```sql
PRIMARY KEY(regexp)
SOURCE(YAMLRegExpTree(PATH '/var/lib/clickhouse/user_files/regexp_tree.yaml'))
```

Is it possible to use this feature from XML at all currently?

**Comment by hanfei1991** [comment]
Yes we cannot create YAMLRegExpTree dictionary with XML config. Because it makes no sense to define structure in the xml config for this dictionary.

**Comment by hkrutzer** [comment]
Why not
```
<source>
  <file>
    <path>/var/lib/clickhouse/user_files/regexp_tree.yaml</path>
  </file>
</source>
```
?
